tmt/log.py

```diff
@@ -0,0 +1,599 @@
+
+        :param logger_name: optional name for the underlying :py:class:`logging.Logger` instance.
+            Useful for debugging. If not set, a generic one is created.
+        :param extra_shift: by how many extra levels should messages be indented by new logger.
+        """
+
+        logger_name = logger_name or f'logger{next(self._child_id_counter)}'
+        actual_logger = self._normalize_logger(self._logger.getChild(logger_name))
+
+        return Logger(
+            actual_logger,
+            base_shift=self._base_shift + extra_shift,
+            verbosity_level=self.verbosity_level,
+            debug_level=self.debug_level,
+            quiet=self.quiet
+            )
+
+    def add_logfile_handler(self, filepath: str) -> None:
+        """ Attach a log file handler to this logger """
+
+        handler = LogfileHandler(filepath)
+
+        handler.setFormatter(LogfileFormatter())
+
+        self._logger.addHandler(handler)
+
+    def add_console_handler(self, apply_colors: bool = False) -> None:
+        """ Attach console handler to this logger """
+
+        handler = ConsoleHandler(stream=sys.stderr)
+
+        handler.setFormatter(ConsoleFormatter(apply_colors=apply_colors))
+
+        handler.addFilter(VerbosityLevelFilter())
+        handler.addFilter(DebugLevelFilter())
+        handler.addFilter(QuietnessFilter())
+
+        self._logger.addHandler(handler)
+
+    def apply_verbosity_options(self, **kwargs: Any) -> 'Logger':
+        """
+        Update logger's settings to match given CLI options.
+
+        Use this method to update logger's settings after :py:meth:`Logger.descend` call,
+        to reflect options given to a tmt subcommand.
+        """
+
+        verbosity_level = cast(Optional[int], kwargs.get('verbose', None))
+        if verbosity_level is None:
+            pass
+
+        elif verbosity_level == 0:
+            pass
+
+        else:
+            self.verbosity_level = verbosity_level
+
+        debug_level_from_global_envvar = _debug_level_from_global_envvar()
+
+        if debug_level_from_global_envvar not in (None, 0):
+            self.debug_level = debug_level_from_global_envvar
+
+        else:
+            debug_level_from_option = cast(Optional[int], kwargs.get('debug', None))
+
+            if debug_level_from_option is None:
+                pass
+
+            elif debug_level_from_option == 0:
+                pass
+
+            else:
+                self.debug_level = debug_level_from_option
+
+        quietness_level = kwargs.get('quiet', False)
+
+        if quietness_level is True:
+            self.quiet = quietness_level
+
+        return self
+
+    @classmethod
+    def create(
+            cls,
+            raw_logger: Optional['Logger'] = None,
+            **verbosity_options: Any) -> 'Logger':
+        if raw_logger is None:
+            actual_logger = cls._normalize_logger(logging.getLogger('tmt'))
+
+        else:
+            actual_logger = raw_logger._logger
```

Comment:
How is this supposed to work when my app has its own logger and I want to wrap it? What do I miss?

```python
import logging
import tmt.log

my_logger = logging.getLogger('my_cool_app')
>>> tmt.log.Logger.create(my_logger)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../site-packages/tmt/log.py", line 476, in create
    actual_logger = raw_logger._logger
AttributeError: 'Logger' object has no attribute '_logger'
```

FTR `tmt.log.Logger(my_logger)` works.

Reply:
Hopefully fixed in ab7486714b177845ff1494b5433cc29ca38f1982

Reply:
Seems to be working fine now.